File: fibonacci/cxx/06_general/test_fibonacci.cpp
#include <catch2/catch_test_macros.hpp>
#include <tuple>
#include <vector>
#include "fibonacci.hpp"

TEST_CASE("fibonacci", "test") {
  std::vector<std::tuple<int, int>> cases {{0, 0}, {1, 1}, {2, 1}, {3, 2}, {10, 55}};
  for (auto c : cases) {
    REQUIRE(fibonacci(std::get<0>(c)) == std::get<1>(c));
  }
}

File: fibonacci/julia/08_linear/test_fibonacci.jl
using Test
include("fibonacci.jl")

@testset "fibonacci" begin
    let testcases = ((-2, -1), (-1, 1), (0, 0), (1, 1), (2, 1), (3, 2), (10, 55), (40, 102334155))
        for (n, v) in testcases
            @test fibonacci(n) == v
        end
    end
end
